graphql_compiler/schema/sqlalchemy_schema.py

```diff
@@ -1,17 +1,92 @@
+import sqlalchemy
+import six
+
+from graphql import GraphQLList
+from graphql.type.definition import GraphQLInterfaceType, GraphQLObjectType, GraphQLUnionType
 
 
 # Complete schema information sufficient to compile GraphQL queries to SQLAlchemy
 #
-# It describes the tables that correspond to each type, and gives instructions on how
-# to perform joins for each vertex field. The property fields on each type are implicitly
-# mapped to columns with the same name on the corresponding table.
+# It describes the tables that correspond to each type (object type, interface type or union type),
+# and gives instructions on how to perform joins for each vertex field. The property fields on each
+# type are implicitly mapped to columns with the same name on the corresponding table.
+#
+# NOTES:
+# - RootSchemaQuery is a special type that does not need a corresponding table.
+# - Builtin types like __Schema, __Type, etc. don't need corresponding tables.
+# - Builtin fields like _x_count do not need corresponding columns.
 SQLAlchemySchemaInfo = namedtuple('SQLAlchemySchemaInfo', (
     'schema',     # GraphQLSchema
-    'tables',     # dict mapping every graphql object type name in the schema to a sqlalchemy table
-    'junctions',  # dict mapping every graphql object type name in the schema to:
+    'tables',     # dict mapping every graphql object type or interface type name in the schema to
+                  # a sqlalchemy table
+    'junctions',  # dict mapping every graphql object type or interface type name in the schema to:
                   #    dict mapping every vertex field name at that type to a dict with keys:
                   #        from_column_name: string, column name to join from
                   #        to_column_name: string, column name to join to
 ))
+
+
+def make_sqlalchemy_schema_info(schema, tables, junctions, validate=True):
+    """Make a SQLAlchemySchemaInfo if the input provided is valid.
+
+    Args:
+        schema: GraphQLSchema
+        tables: dict mapping every graphql object type or interface type name in the schema to
+                a sqlalchemy table
+        junctions: dict mapping every graphql object type or interface type name in the schema to:
+                       dict mapping every vertex field name at that type to a dict with keys:
+                           from_column_name: string, column name to join from
+                           to_column_name: string, column name to join to
+        validate: Optional bool (default True), specifying whether to validate that the given
+                  input is valid for creation of a SQLAlchemySchemaInfo. Consider not validating
+                  to save on performance when dealing with a large schema.
+
+    Returns:
+        SQLAlchemySchemaInfo containing the input arguments provided
+    """
+    if validate:
+        types_mapped_to_tables = (GraphQLInterfaceType, GraphQLObjectType, GraphQLUnionType)
+        builtin_types = {
```

Comment:
For the sake of future-proofing this, I think we should mark all types prefixed with `__` as built-in, since the entire `__` (double-underscore) prefix is reserved for GraphQL internal use.